### ISSUE TITLE
GraalVM Data test failure: should not reach here: class net.bytebuddy.build.HashCodeAndEqualsPlugin

### ISSUE THREAD

**Issue by sdelamo** [micronaut-projects/micronaut-core#7085]
### Issue description

Steps to reproduce: 

- Clone [micronaut-data-jpa-graal](https://github.com/micronaut-graal-tests/micronaut-data-jpa-graal)
- switch to branch '3.4.x_mariadb`
- `./build-native-image.sh`

https://gitlab.com/micronaut-projects/micronaut-graal-tests/-/jobs/2208558955

```
Fatal error: org.graalvm.compiler.debug.GraalError: should not reach here: class net.bytebuddy.build.HashCodeAndEqualsPlugin$Enhance$InvokeSuper$1
	at jdk.internal.vm.compiler/org.graalvm.compiler.debug.GraalError.shouldNotReachHere(GraalError.java:51)
	at com.oracle.svm.reflect.hosted.ReflectionMetadataEncoderImpl.tag(ReflectionMetadataEncoderImpl.java:[116](https://gitlab.com/micronaut-projects/micronaut-graal-tests/-/jobs/2208558955#L116)2)
	at com.oracle.svm.reflect.hosted.ReflectionMetadataEncoderImpl.encodeValue(ReflectionMetadataEncoderImpl.java:1037)
	at com.oracle.svm.reflect.hosted.ReflectionMetadataEncoderImpl.encodeMemberValue(ReflectionMetadataEncoderImpl.java:1032)
	at com.oracle.svm.reflect.hosted.ReflectionMetadataEncoderImpl.encodeAllAndInstall(ReflectionMetadataEncoderImpl.java:767)
	at com.oracle.svm.hosted.image.NativeImageCodeCache.buildRuntimeMetadata(NativeImageCodeCache.java:326)
	at com.oracle.svm.hosted.image.LIRNativeImageCodeCache.layoutMethods(LIRNativeImageCodeCache.java:106)
	at com.oracle.svm.hosted.NativeImageGenerator.doRun(NativeImageGenerator.java:640)
	at com.oracle.svm.hosted.NativeImageGenerator.run(NativeImageGenerator.java:514)
	at com.oracle.svm.hosted.NativeImageGeneratorRunner.buildImage(NativeImageGeneratorRunner.java:413)
	at com.oracle.svm.hosted.NativeImageGeneratorRunner.build(NativeImageGeneratorRunner.java:591)
	at com.oracle.svm.hosted.NativeImageGeneratorRunner.main(NativeImageGeneratorRunner.java:129)
	at com.oracle.svm.hosted.NativeImageGeneratorRunner$JDK9Plus.main(NativeImageGeneratorRunner.java:621)
```

**Comment by graemerocher** [comment]
I am looking into this one

**Comment by graemerocher** [comment]
I can't reproduce this with the current stable 22.0.0 release of GraalVM

**Comment by sdelamo** [comment]
@graemerocher Data tests fail now for a different reason for 3.4.x and latest master commit of GraalVM. 

❌ https://gitlab.com/micronaut-projects/micronaut-graal-tests/-/jobs/2249088587


```
o.micronaut.context.exceptions.BeanInstantiationException: Bean definition [org.hibernate.SessionFactory] could not be loaded: Error instantiating bean of type [org.hibernate.SessionFactory]: Unable to load TypeSafeActivator class
[65](https://gitlab.com/micronaut-projects/micronaut-graal-tests/-/jobs/2249088587#L65)	at io.micronaut.context.DefaultBeanContext.initializeContext(DefaultBeanContext.java:1938)
[66](https://gitlab.com/micronaut-projects/micronaut-graal-tests/-/jobs/2249088587#L66)	at io.micronaut.context.DefaultApplicationContext.initializeContext(DefaultApplicationContext.java:237)
[67](https://gitlab.com/micronaut-projects/micronaut-graal-tests/-/jobs/2249088587#L67)	at io.micronaut.context.DefaultBeanContext.readAllBeanDefinitionClasses(DefaultBeanContext.java:3453)
[68](https://gitlab.com/micronaut-projects/micronaut-graal-tests/-/jobs/2249088587#L68)	at io.micronaut.context.DefaultBeanContext.finalizeConfiguration(DefaultBeanContext.java:3883)
[69](https://gitlab.com/micronaut-projects/micronaut-graal-tests/-/jobs/2249088587#L69)	at io.micronaut.context.DefaultBeanContext.start(DefaultBeanContext.java:329)
[70](https://gitlab.com/micronaut-projects/micronaut-graal-tests/-/jobs/2249088587#L70)	at io.micronaut.context.DefaultApplicationContext.start(DefaultApplicationContext.java:183)
[71](https://gitlab.com/micronaut-projects/micronaut-graal-tests/-/jobs/2249088587#L71)	at io.micronaut.runtime.Micronaut.start(Micronaut.java:[72](https://gitlab.com/micronaut-projects/micronaut-graal-tests/-/jobs/2249088587#L72))
72	at io.micronaut.runtime.Micronaut.run(Micronaut.java:320)
[73](https://gitlab.com/micronaut-projects/micronaut-graal-tests/-/jobs/2249088587#L73)	at io.micronaut.runtime.Micronaut.run(Micronaut.java:306)
[74](https://gitlab.com/micronaut-projects/micronaut-graal-tests/-/jobs/2249088587#L74)	at example.Application.main(Application.java:30)
[75](https://gitlab.com/micronaut-projects/micronaut-graal-tests/-/jobs/2249088587#L75)Caused by: io.micronaut.context.exceptions.BeanInstantiationException: Error instantiating bean of type [org.hibernate.SessionFactory]: Unable to load TypeSafeActivator class
[76](https://gitlab.com/micronaut-projects/micronaut-graal-tests/-/jobs/2249088587#L76)	at io.micronaut.context.DefaultBeanContext.doCreateBean(DefaultBeanContext.java:2365)
[77](https://gitlab.com/micronaut-projects/micronaut-graal-tests/-/jobs/2249088587#L77)	at io.micronaut.context.DefaultBeanContext.createAndRegisterSingletonInternal(DefaultBeanContext.java:3281)
[78](https://gitlab.com/micronaut-projects/micronaut-graal-tests/-/jobs/2249088587#L78)	at io.micronaut.context.DefaultBeanContext.createAndRegisterSingleton(DefaultBeanContext.java:3267)
[79](https://gitlab.com/micronaut-projects/micronaut-graal-tests/-/jobs/2249088587#L79)	at io.micronaut.context.DefaultBeanContext.loadContextScopeBean(DefaultBeanContext.java:2653)
[80](https://gitlab.com/micronaut-projects/micronaut-graal-tests/-/jobs/2249088587#L80)	at io.micronaut.context.DefaultBeanContext.initializeContext(DefaultBeanContext.java:1932)
[81](https://gitlab.com/micronaut-projects/micronaut-graal-tests/-/jobs/2249088587#L81)	... 9 common frames omitted
[82](https://gitlab.com/micronaut-projects/micronaut-graal-tests/-/jobs/2249088587#L82)Caused by: org.hibernate.HibernateException: Unable to load TypeSafeActivator class
[83](https://gitlab.com/micronaut-projects/micronaut-graal-tests/-/jobs/2249088587#L83)	at org.hibernate.cfg.beanvalidation.BeanValidationIntegrator.loadTypeSafeActivatorClass(BeanValidationIntegrator.java:195)
[84](https://gitlab.com/micronaut-projects/micronaut-graal-tests/-/jobs/2249088587#L84)	at org.hibernate.cfg.beanvalidation.BeanValidationIntegrator.integrate(BeanValidationIntegrator.java:111)
[85](https://gitlab.com/micronaut-projects/micronaut-graal-tests/-/jobs/2249088587#L85)	at org.hibernate.internal.SessionFactoryImpl.<init>(SessionFactoryImpl.java:292)
[86](https://gitlab.com/micronaut-projects/micronaut-graal-tests/-/jobs/2249088587#L86)	at org.hibernate.boot.internal.SessionFactoryBuilderImpl.build(SessionFactoryBuilderImpl.java:471)
[87](https://gitlab.com/micronaut-projects/micronaut-graal-tests/-/jobs/2249088587#L87)	at io.micronaut.configuration.hibernate.jpa.EntityManagerFactoryBean.hibernateSessionFactory(EntityManagerFactoryBean.java:222)
[88](https://gitlab.com/micronaut-projects/micronaut-graal-tests/-/jobs/2249088587#L88)	at io.micronaut.configuration.hibernate.jpa.$EntityManagerFactoryBean$HibernateSessionFactory3$Definition.build(Unknown Source)
[89](https://gitlab.com/micronaut-projects/micronaut-graal-tests/-/jobs/2249088587#L89)	at io.micronaut.context.BeanDefinitionDelegate.build(BeanDefinitionDelegate.java:136)
[90](https://gitlab.com/micronaut-projects/micronaut-graal-tests/-/jobs/2249088587#L90)	at io.micronaut.context.DefaultBeanContext.doCreateBean(DefaultBeanContext.java:2336)
[91](https://gitlab.com/micronaut-projects/micronaut-graal-tests/-/jobs/2249088587#L91)	... 13 common frames omitted
[92](https://gitlab.com/micronaut-projects/micronaut-graal-tests/-/jobs/2249088587#L92)Caused by: org.hibernate.boot.registry.classloading.spi.ClassLoadingException: Unable to load class [org.hibernate.cfg.beanvalidation.TypeSafeActivator]
[93](https://gitlab.com/micronaut-projects/micronaut-graal-tests/-/jobs/2249088587#L93)	at org.hibernate.boot.registry.classloading.internal.ClassLoaderServiceImpl.classForName(ClassLoaderServiceImpl.java:133)
[94](https://gitlab.com/micronaut-projects/micronaut-graal-tests/-/jobs/2249088587#L94)	at org.hibernate.cfg.beanvalidation.BeanValidationIntegrator.loadTypeSafeActivatorClass(BeanValidationIntegrator.java:192)
[95](https://gitlab.com/micronaut-projects/micronaut-graal-tests/-/jobs/2249088587#L95)	... 20 common frames omitted
[96](https://gitlab.com/micronaut-projects/micronaut-graal-tests/-/jobs/2249088587#L96)Caused by: java.lang.ClassNotFoundException: org.hibernate.cfg.beanvalidation.TypeSafeActivator
[97](https://gitlab.com/micronaut-projects/micronaut-graal-tests/-/jobs/2249088587#L97)	at java.lang.Class.forName(DynamicHub.java:1081)
[98](https://gitlab.com/micronaut-projects/micronaut-graal-tests/-/jobs/2249088587#L98)	at org.hibernate.boot.registry.classloading.internal.ClassLoaderServiceImpl.classForName(ClassLoaderServiceImpl.java:130)
[99](https://gitlab.com/micronaut-projects/micronaut-graal-tests/-/jobs/2249088587#L99)	... 21 common frames omitted
[100](https://gitlab.com/micronaut-projects/micronaut-graal-tests/-/jobs/2249088587#L100)+ curl -s localhost:8080/owners
```



They pass for 22.0.0 release of GraalVM and 2.4.x: 
✅ https://gitlab.com/micronaut-projects/micronaut-graal-tests/-/jobs/2248810752

**Comment by sdelamo** [comment]
This is now fixed for 2.4.x-dev. There is [another issue](https://github.com/micronaut-projects/micronaut-core/issues/7147).
